scripts/import_from_kc.py

```diff
@@ -0,0 +1,260 @@
+        WHERE type = 'LOGIN'
+        GROUP BY user_id, client_id
+        """
+    )
+    app_links_data = cursor.fetchall()
+    users_app_links = defaultdict(list)
+    for user_id, client_id, event_time in app_links_data:
+        users_app_links[user_id].append([client_id, parse_keycloak_dt(event_time)])
+
+    # last_login
+    cursor.execute(
+        """
+        SELECT user_id, MAX(event_time)
+        FROM event_entity
+        WHERE type = 'LOGIN'
+        GROUP BY user_id
+        """
+    )
+    last_login_data = cursor.fetchall()
+    users_last_login = {}
+    for user_id, event_time in last_login_data:
+        users_last_login[user_id] = parse_keycloak_dt(event_time)
+
+    # Applications
+    cursor.execute(
+        """
+        SELECT client_id, secret, client.name, realm.name
+        FROM client
+        INNER JOIN realm ON client.realm_id = realm.id
+        WHERE secret is not NULL
+        """
+    )
+    application_data = cursor.fetchall()
+
+    # Redirect uris
+    cursor.execute(
+        """
+        SELECT client.client_id, redirect_uris.value
+        FROM redirect_uris
+        INNER JOIN client
+        ON client.id = redirect_uris.client_id
+        """
+    )
+    redirect_uris_data = cursor.fetchall()
+    redirect_uris = defaultdict(str)
+    for client_id, value in redirect_uris_data:
+        redirect_uris[client_id] += f",{value}"
+
+applications = {}
+for client_id, secret, name, realm_name in application_data:
+    if realm_name not in REAMLS:
+        continue
+    applications[client_id] = Application(
+        client_id=client_id,
+        client_type="confidential",
+        authorization_grant_type="authorization-code",
+        client_secret=secret,
+        name=name,
+        algorithm="RS256",
+        redirect_uris=redirect_uris.get(client_id, ",")[1:],  # Remove initial comma
+    )
+
+users = []
+email_addresses = []
+app_links = []
+for user_id, username, email, first_name, last_name, email_verified, created_timestamp, realm_name in users_data:
+    if realm_name not in REAMLS:
+        continue
+    created_at = parse_keycloak_dt(created_timestamp)
+    email_verified = email_verified and not user_id in users_must_verify_email
+    if not email_verified and created_at < timezone.now() - datetime.timedelta(days=UNVERIFED_USERS_DAYS_CUTOFF):
+        continue
+    user = User(
+        username=user_id,
+        email=email if email_verified else "",
+        first_name=first_name,
+        last_name=last_name,
+        date_joined=created_at,
+        last_login=users_last_login.get(user_id),
+        password=credentials.get(user_id, ""),
+        must_reset_password=user_id in users_must_reset_password,
+        terms_accepted_at=None if user_id in users_must_accept_terms else created_at,
+    )
+    if email_verified:
+        email_addresses.append(EmailAddress(user=user, email=email, verified_at=created_at))
+
+    for client_id, last_login in users_app_links.get(user_id, []):
```

Comment:
C‘est un `defaultdict`, pas besoin de `.get()` :)
```suggestion
    for client_id, last_login in users_app_links[user_id]:
```